### DISCUSSION TITLE
Passing URL query parameters from a preview to the component (params[:foo])

### DISCUSSION THREAD

**Discussion by fredboyle** [lookbook-hq/lookbook#580] [Q&A]
Trying to figure out how to replicate the Rails `params` hash in a standard view but in a Lookbook preview.

The preview classes don't receive a request or params object so how can we retrieve the query params from the URL in a Lookbook preview?

This is to enable functionality in components, such as pagination, that pass values to the URL and reload the page.

Thanks for your guidance.

**Comment by Spone** [comment]
Hi @fredboyle have you tried [Dynamic params](https://lookbook.build/guide/previews/params)? It may help you to achieve what you're looking for.

**Comment by fredboyle** [comment]
@Spone I thought of that but was hoping for something more generic and less Lookbook-specific. 

Additionally, the parameter name is configurable so is dynamic itself.

If there's nothing other an dynamic params than it'll have to do.
